test/functional/legacy/062_tab_pages_spec.lua

```diff
@@ -0,0 +1,272 @@
+-- Tests for tab pages
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source, clear, execute, expect, eval, eq =
+  helpers.feed, helpers.insert, helpers.source, helpers.clear,
+  helpers.execute, helpers.expect, helpers.eval, helpers.eq
+
+describe('tab pages', function()
+  setup(clear)
+
+  it('is working', function()
+    insert('Results:')
+
+    execute('lang C')
+
+    -- Simple test for opening and closing a tab page.
+    execute('tabnew')
+    eq(2, eval('tabpagenr()'))
+    execute('q')
+    eq(1, eval('tabpagenr()'))
+
+    -- Open three tab pages and use ":tabdo".
+    source([[
+      0tabnew
+      1tabnew
+      $tabnew
+      tabdo call append(line('$'), 'this is tab page ' . tabpagenr())
+      tabclose! 2
+      tabrewind
+    ]])
+    eq('this is tab page 1', eval("getline('$')"))
+    execute('undo')
+    execute('q')
+    execute('tablast')
+    eq('this is tab page 4', eval("getline('$')"))
+    execute('q!')
+
+    -- Test for settabvar() and gettabvar() functions. Open a new tab page and
+    -- set 3 variables to a number, string and a list. Verify that the
+    -- variables are correctly set.
+    source([[
+      tabnew
+      tabfirst
+      call settabvar(2, 'val_num', 100)
+      call settabvar(2, 'val_str', 'SetTabVar test')
+      call settabvar(2, 'val_list', ['red', 'blue', 'green'])
+    ]])
+
+    eq(100, eval('gettabvar(2, "val_num")'))
+    eq('SetTabVar test', eval('gettabvar(2, "val_str")'))
+    eq({'red', 'blue', 'green'}, eval('gettabvar(2, "val_list")'))
+    execute('tabnext 2')
+    eq(100, eval('t:val_num'))
+    eq('SetTabVar test', eval('t:val_str'))
+    eq({'red', 'blue', 'green'}, eval('t:val_list'))
+    execute('tabclose')
+
+    -- Test some drop features
+    -- Test for ":tab drop exist-file" to keep current window.
+    execute('sp test1')
+    execute('tab drop test1')
+    eq(1, eval('tabpagenr("$")'))
+    eq(2, eval('winnr("$")'))
+    eq(1, eval('winnr()'))
+    execute('close')
+    -- Test for ":tab drop new-file" to keep current window of tabpage 1.
+    execute('split')
+    execute('tab drop newfile')
+    eq(2, eval('tabpagenr("$")'))
+    eq(2, eval('tabpagewinnr(1, "$")'))
+    eq(1, eval('tabpagewinnr(1)'))
+    execute('tabclose')
+    execute('q')
+
+    -- Test for ":tab drop multi-opend-file" to keep current tabpage and
+    -- window.
+    execute('new test1')
+    execute('tabnew')
+    execute('new test1')
+    execute('tab drop test1')
+    eq(2, eval('tabpagenr()'))
+    eq(2, eval('tabpagewinnr(2, "$")'))
+    eq(1, eval('tabpagewinnr(2)'))
+    execute('tabclose')
+    execute('q')
```

Comment:
Could these tests be split up into `it()` blocks? I think it would avoid the need for `execute('q')` and/or `execute('tabclose')`, but I'm not too experienced with these tests so idk.